backend/api/server.js

```diff
@@ -45,6 +46,259 @@ server.post('/login', async (req, res, next) => {
+    const patient = await prisma.patient.findUnique({
+        where: { id: patientId },
+        include: {
+            provider: {
+                include: {
+                    user: true
+                }
+            }
+        }
+    });
+
+    if (!patient) {
+        return res.status(204).json(`No patient with id: ${patientId}`);
+    }
+
+    if (!patient.provider) {
+        return res.status(204).json(`No patient has no provider!`);
+    }
+
+    return res.status(200).json(patient.provider);
+});
+
+server.put('/patients/:patientId/provider', async (req, res, next) => {
+    const patientId = Number(req.params.patientId);
+    const { providerId } = req.body;
+
+    const patient = await prisma.patient.update({
+        where: { id: patientId },
+        include: { user: true },
+        data: { provider_id: providerId }
+    });
+
+    if (!patient) {
+        return res.status(204).json(`No patient with id: ${patientId}`);
+    }
+
+    patient.provider_id = provider_id;
+    return res.status(200).json(patient);
+});
+
+
+/* --- Provider Endpoints --- */
+
+server.get('/providers/:providerId', async (req, res, next) => {
+    const providerId = Number(req.params.providerId);
+    const provider = await prisma.provider.findUnique({
+        where: { id: providerId },
+        include: { user: true }
+    });
+
+    if (!provider) {
+        return res.status(204).json(`No provider with id: ${providerId}`);
+    }
+
+    return res.status(200).json(provider);
+});
+
+server.get('/providers/:providerId/patients', async (req, res, next) => {
+    const providerId = Number(req.params.providerId);
+    const provider = await prisma.provider.findUnique({
+        where: { id: providerId },
+        include: { patients: true }
+    });
+
+    if (!provider) {
+        return res.status(204).json(`No provider with id: ${providerId}`);
+    }
+
+    return res.status(200).json(provider);
+});
+
+
+/* --- Medication Endpoints --- */
+
+server.get('/patients/:patientId/medications', async (req, res, next) => {
+    const patientId = Number(req.params.patientId);
+    const patient = await prisma.patient.findUnique({
+        where: { id: patientId },
+        include: { medications: true }
+    });
+
+    if (!patient) {
+        return res.status(204).json(`No patient with id: ${patientId}`);
+    }
+
+    return res.status(200).json(patient.medications);
+});
+
+server.get('/patients/:patientId/medications/:medicationId', async (req, res, next) => {
+    const patientId = Number(req.params.patientId);
+    const medicationId = Number(req.params.medicationId);
+    const medication = await prisma.medication.findUnique({
+        where: { id: medicationId },
+    });
+
+    if (!medication) {
+        return res.status(204).json(`No medication with id: ${medicationId}`);
+    }
+
+    if (medication.patient_id != patientId) {
+        return res.status(204).json(`No medication belonging to patient with id: ${patientId}`)
```

Comment:
A bit different text here as they are specifying the medicationId
```suggestion
        return res.status(204).json(`This medication (${medicationId}) does not belong to patient with id: ${patientId}`)
```

Reply:
Makes sense! Thank you!